adb/filesync_protocol.py

```diff
@@ -159,7 +158,7 @@ def Send(self, command_id, data=b'', size=0):
       size: Optionally override size from len(data).
     """
     if data:
-      if not isinstance(data, bytes):
+      if isinstance(data, str):
```

Comment:
Strings are also bytes in Python3. Using str here instead for py2/3 compatibility then always encoding as bytes.

Reply:
With this change, unicode in Python won't be encoded and we'll try to send unicode over the wire.

What was wrong with the previous line? We need bytes, so if it's not bytes, we encode using utf8 to get bytes.

Reply:
Fixed